### ISSUE TITLE
Approval queues for groups

### ISSUE THREAD

**Issue by sweetaibelle** [knighty/fimfiction-issues#106]
I figure since there is an issue tracker now, this'd be a good place to suggest things.

One thing a number of groups like to do is make it so that incoming stories in the group have to be approved before they are allowed in. There isn't really any functionality for that in place at the moment.

One of the more common things to do is the group will have one folder labeled "Incoming" that everyone can put stories in, and then all the other ones are contributor level or higher. This _works_, but not really that well.

For one thing, if you make approvers contributors, they can add stories to the other folders, but they won't be able to delete stories out of the incoming folder. You have to be a group admin for that. So you either need to make them all admins, or have an admin going behind them deleting everything that's been approved. (Admittedly, making what contributors can do more customizable by the group admins would go a long way to help, too. Right now they aren't really that useful.)

Another thing is that it's easy to cherry pick stories rather than grabbing the first one off the top if it's just a folder.

There are other niceties, too, like depending on how it was implemented, you could make it where the person submitting it had already put it in the folders it was intended for, so that the approver didn't have to.

This obviously isn't anything critical, just a low priority enhancement suggestion, but there are a lot of review type groups it'd be useful in. And who knows, maybe you could reuse some of the site story approval code.

_--Sweetie Belle_

**Comment by ssokolow** [comment]
It'd also be nice if, as a user, I could choose whether stories pending approval show up in my feed.

As-is, I just have to ignore the feed entries for incoming folders in groups where I only want the stuff that meets the group's standards.

(But it should be an option I can toggle on a per-group basis. There are actually various cases where I've quite enjoyed stories that ended up not making the cut because they weren't quite on topic for the group and I'd never have found them otherwise.)
